### ISSUE TITLE
Error during compilation

### ISSUE THREAD

**Issue by tittoassini** [robinpowered/react-native-device-battery#1]
/Users/titto/workspace/babbo/android/app/src/main/java/com/babbo/MainActivity.java:34: error: constructor DeviceBatteryPackage in class DeviceBatteryPackage cannot be applied to given types;
                .addPackage(new DeviceBatteryPackage())
                            ^
  required: Activity
  found: no arguments
  reason: actual and formal argument lists differ in length
1 error
:app:compileDebugJavaWithJavac FAILED

**Comment by happyEgg** [comment]
I know the reason, you need to modify MainActivity.java，
@Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
        new MainReactPackage(),
        new RNSeekBarPackage(),
        new RNFSPackage(),
        new DeviceBatteryPackage(this)
      );

add "new DeviceBatteryPackage(this)"

**Comment by atticoos** [comment]
Thanks for taking a look @happyEgg, that appears to be precisely the issue. 

Upon reviewing our instructions, it appears that we aren't showing this argument

**Comment by manuelm3z** [comment]
Hey, I'm having trouble using this package, but I think it's because of the react native version, I just want to know if you're planning to support 0.43

**Comment by atticoos** [comment]
@manuelm3z is this on iOS or Android, and do you have any information about the compilation error you can share?

**Comment by manuelm3z** [comment]
@ajwhite i'm trying on android, using react-native 0.43 and the error is this

![Error](https://i.gyazo.com/57b1cf899af506c7929d1b2b1bae4797.png "Error")

Looking at the issues I found this about compilation errors, I tried the change but it did not work, then I understood that the package needs an Activity and in this version of react-native the package is instantiated MainApplication.java instead of MainActivity.java, In that case I'm not really sure what to do.

![Code](https://i.gyazo.com/10423b18193202b6c9e12eb797bb5826.png "MainApplication.java")

**Comment by atticoos** [comment]
@manuelm3z Ah, good catch. It looks like there's been some work on this front, but an official release hasn't been made.

Would you mind installing this dependency from the repository against the commit b7cc53a51810389c226efec8da9e45a38e49c2ca?

```
npm install git+ssh://git@github.com/robinpowered/react-native-device-battery.git#b7cc53a51810389c226efec8da9e45a38e49c2ca
```

**Comment by manuelm3z** [comment]
@ajwhite  Great, it works. Thank you.

**Comment by moritzwachter** [comment]
@ajwhite Are there any plans to update the npm package in the near future?

**Comment by atticoos** [comment]
The changes that solved this have since been published under `v1.0.0`
